src/foam/core/ThreadTimerTask.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * @license
+ * Copyright 2023 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.core;
+
+import foam.core.ContextAgent;
+import foam.core.ProxyX;
+import foam.core.X;
+import foam.core.XLocator;
+import java.util.TimerTask;
+
+/**
+  TimerTask which executes ContextAgent in it's own thread.
+*/
+public class ThreadTimerTask
+  extends TimerTask {
+
+  protected X x_;
+  protected ContextAgent agent_;
+  protected String name_;
+  public int priority_ = Thread.MAX_PRIORITY;
```

Comment:
Do we want to do this? Could it starve other threads? Would default be sufficient?